### ISSUE TITLE
replace-text-with-dimensions

### ISSUE THREAD

**Issue by organicweb** [Compass/compass#1611]
Hello, I'm running compass and compass-rails in a rails 4.0.3 project
compass 0.12.3
compass-rails 1.1.7
just find out that : 

```
+replace-text("logo.png", 0px, 0px)
+replace-text(image-url("logo.png"), 0px, 0px)
```

both works like a charm, but

```
+replace-text-with-dimensions(image-url("logo.png"), 0px, 0px, true)
```

return an error like this : 

```
File not found or cannot be read: /Users/mathias/Documents/Clients/Sgar35/fete-de-europe-2014/fde-2014/app/assets/images/url(/assets/logo.png)
  (in /Users/mathias/Documents/Clients/Sgar35/fete-de-europe-2014/fde-2014/app/assets/stylesheets/screen.css.sass)
```

and not

```
+replace-text-with-dimensions("logo.png", 0px, 0px, true)
```

hope it helps

**Comment by chriseppstein** [comment]
You're passing a url to be inlined? That doesn't make much sense. This should give a better error.

**Comment by organicweb** [comment]
Sorry it's a mistake inline was false at this time of test.
Just did the test again not inlined, and it's not working properly.

**Comment by chriseppstein** [comment]
Please provide an alternate description if you think the description above is incorrect. I looked at the code and it very clearly handles the non-inline case because it delegates to replace-text.

**Comment by organicweb** [comment]
I'm sorry but it does not handle this case : 

```
+replace-text-with-dimensions(image-url("logo.png"), 0px, 0px)

```

It returns : 

```


Showing /Users/mathias/Documents/Clients/Sgar35/fete-de-europe-2014/fde-2014/app/views/layouts/application.html.haml where line #14 raised:

File not found or cannot be read: /Users/mathias/Documents/Clients/Sgar35/fete-de-europe-2014/fde-2014/app/assets/images/url(/assets/logo.png)
  (in /Users/mathias/Documents/Clients/Sgar35/fete-de-europe-2014/fde-2014/app/assets/stylesheets/screen.css.sass)

```

Seems to have a glitch somewhere, I saw too in code that it's handled by replace-text method.

**Comment by chriseppstein** [comment]
It cannot find the file, so it cannot put the dimensions of the file in your stylesheet. That's an error.

If you want the dimensions you need to not pass the url. This should be a more clear error.

**Comment by organicweb** [comment]
Please take a proper look at the path of the unreadable file !!!

It's not properly build : /app/assets/images/url(/assets/logo.png

**Comment by mirisuzanne** [comment]
You don't need to use the image-url() function with this mixin. Try:

``` sass
+replace-text-with-dimensions("logo.png", 0px, 0px)
```

**Comment by organicweb** [comment]
Yeah, I know I was just telling you with image-url it's not working that's all…
As it's handled by the replace-text method it should work properly

No offense to anyone

**Comment by chriseppstein** [comment]
The only issue here is that the error is not very clear. There's no reason to expect that passing in a url reference to this mixin should work. Are you expecting that compass would download the image or maybe that it would reverse engineer the url and realize it's an image in the local filesystem?

Anyways. it's easy to give a better error message, I can see how someone might inadvertently do this without thinking.
